static/js/components/tiles/answer_table_tile.tsx

```diff
@@ -0,0 +1,177 @@
+  entities: string[];
+  // Columns in the table
+  columns: AnswerTableColumn[];
+}
+
+interface AnswerTableData {
+  entityNames: Record<string, string>;
+  values: Record<string, Record<string, string[]>>;
+  sources: string[];
+}
+
+export function AnswerTableTile(props: AnswerTableTilePropType): JSX.Element {
+  const [answerData, setAnswerData] = useState<AnswerTableData>(null);
+
+  useEffect(() => {
+    fetchData(props).then((data) => {
+      setAnswerData(data);
+    });
+  }, [props]);
+
+  if (!answerData) {
+    return null;
+  }
+
+  return (
+    <div
+      className={`chart-container answer-table-tile ${ASYNC_ELEMENT_HOLDER_CLASS}`}
+    >
+      <div className={`answer-table-content ${ASYNC_ELEMENT_CLASS}`}>
+        <div className="answer-table-title">{props.title}</div>
+        <table className="answer-table">
+          <thead>
+            <tr>
+              <th></th>
+              {props.columns.map((col, idx) => {
+                return <th key={`col-header-${idx}`}>{col.header}</th>;
+              })}
+            </tr>
+          </thead>
+          <tbody>
+            {props.entities.map((entity, rowIdx) => {
+              return (
+                <tr key={`row-${rowIdx}`}>
+                  <td>
+                    <a href={URI_PREFIX + entity}>
+                      {answerData.entityNames[entity] || entity}
+                      <span className="material-icons-outlined">
+                        arrow_forward
+                      </span>
+                    </a>
+                  </td>
+                  {props.columns.map((col, colIdx) => {
+                    return (
+                      <td key={`row-${rowIdx}-col-${colIdx}`}>
+                        {answerData.values[entity][col.propertyExpr].join(", ")}
+                      </td>
+                    );
+                  })}
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+      </div>
+      <div className="source">source: {answerData.sources.join(", ")}</div>
+    </div>
+  );
+}
+
+const fetchData = async (
+  props: AnswerTableTilePropType
+): Promise<AnswerTableData> => {
+  const propertyPromises = [];
+  for (const column of props.columns) {
+    propertyPromises.push(
+      axios.get(`/api/node/propvals`, {
+        params: { dcids: [props.entities], propExpr: column.propertyExpr },
+        paramsSerializer: stringifyFn,
+      })
+    );
+  }
+  // Also get the names for the entities
+  propertyPromises.push(
+    axios.get(`/api/node/propvals`, {
+      params: { dcids: [props.entities], propExpr: "->name" },
+      paramsSerializer: stringifyFn,
+    })
+  );
+  try {
+    const propResp = await Promise.all(propertyPromises);
+    const entityNames = {};
+    const values = {};
+    props.entities.forEach((entity) => {
+      values[entity] = {};
+    });
+    const provIds: Set<string> = new Set();
+    propResp.forEach((resp, i) => {
+      if (i >= props.columns.length) {
```

Comment:
Alternatively, what do you think of making the names property fetch its own promise with its own logic, similar to what you did for the providences?

Reply:
The provences is separated out because it depends on the information retrieved in the first fetch. I think it'll be better to do the name fetch at the same time as the other property values so that the fetches can happen in parallel. I updated the code to hopefully make things more clear:
- moved the name fetch to be the first promise
- separated out resp into nameResp and propResp after the fetch & have separated out the logic for handling
- added comments